Client/Frontend/Browser/Tab.swift

```diff
@@ -909,9 +927,8 @@ private class TabContentScriptManager: NSObject, WKScriptMessageHandler {
     }
 
     func addContentScript(_ helper: TabContentScript, name: String, forTab tab: Tab) {
-        if let _ = helpers[name] {
-            assertionFailure("Duplicate helper added: \(name)")
-        }
+        // If a helper script already exists on a tab, skip adding this duplicate.
+        if let _ = helpers[name] { return }
```

Comment:
q1: When we clear all tabs and if the script exists don't we need the script to be re-added for that tab or basically what happens then?

q2: we should make it a guard 😅

Reply:
This is for the case where we reload the tab for special case

Reply:
It looks like when we `close()`, we uninstall all contentScripts. And when we navigate back to the tab (whether it's the currently open one that gets removed/added again or others), these scripts need to get added again. 

This check is just so that we prevent adding a duplicate of the same script. Not sure if that answered your Q 😓 

Also, yes to the guard!